src/Workspaces/Core/Portable/FindSymbols/SymbolTree/SymbolTreeInfo_Metadata.cs

```diff
@@ -17,14 +17,39 @@ namespace Microsoft.CodeAnalysis.FindSymbols
 {
     internal partial class SymbolTreeInfo
     {
-        private static char[] s_dotSeparator = { '.' };
-
         private static string GetMetadataNameWithoutBackticks(MetadataReader reader, StringHandle name)
         {
-            var typeName = reader.GetString(name);
-            var index = typeName.IndexOf('`');
-            typeName = index > 0 ? typeName.Substring(0, index) : typeName;
-            return typeName;
+            var blobReader = reader.GetBlobReader(name);
+            var backtickIndex = IndexOfCharacter(blobReader, '`');
+            if (backtickIndex == -1)
+            {
+                return reader.GetString(name);
+            }
+
+            unsafe
+            {
+                return MetadataStringDecoder.DefaultUTF8.GetString(
+                    blobReader.CurrentPointer, backtickIndex);
+            }
+        }
+
+        private static int IndexOfCharacter(BlobReader blobReader, char ch)
+        {
+            unsafe
```

Comment:
I want to minimize unsafe code.  So i use blocks for that purpose instead of making the entire body unsafe.  Now, in this case, i needed to make the block as large as teh body.  But still, i prefer that :)